lib/sgtn-client/api/translation.rb

```diff
@@ -86,22 +86,30 @@ def self.getTranslation(component, key, locale)
       end
 
       def self.get_cs(component, locale)
+        if locale.nil?
+          locale=SgtnClient::Config.configurations.default
+        end
+        locale = locale.to_s
         flocale = SgtnClient::LocaleUtil.fallback(locale)
         cache_key = SgtnClient::CacheUtil.get_cachekey(component, flocale)
         SgtnClient.logger.debug "[Translation][get_cs]cache_key=#{cache_key}"
         expired, items = SgtnClient::CacheUtil.get_cache(cache_key)
-        if items.nil? || expired
-          items = load(component, flocale)
-          if items.nil?
-            items = SgtnClient::Source.getSources(component, SgtnClient::Config.configurations.default)
-            SgtnClient::Core::Cache.put(cache_key, items, 60)
+        t = Thread.new {
+          if items.nil? || expired 
+            items = load(component, flocale)
+            if items.nil?
+              items = SgtnClient::Source.getSources(component, SgtnClient::Config.configurations.default)
+              SgtnClient::Core::Cache.put(cache_key, items, 60)
+            else
+              SgtnClient::CacheUtil.write_cache(cache_key, items)
+            end
           else
-            SgtnClient::CacheUtil.write_cache(cache_key, items)
+            SgtnClient.logger.debug "[Translation]get translations from cache with key: " + cache_key
```

Comment:
Isn't this supposed to be in a different PR?

Reply:
the logic is not change.

Reply:
Sure, this is a minor change so not a big deal. It relates to this issue though (@lyiyu66 ): #https://github.com/vmware/singleton/issues/1484